src/SFML/Graphics/Shader.cpp

```diff
@@ -40,6 +40,18 @@
 
 #ifndef SFML_OPENGL_ES
 
+#if defined(SFML_SYSTEM_MACOS) || defined(SFML_SYSTEM_IOS)
+
+    #define castToGlHandle(x) reinterpret_cast<GLEXT_GLhandle>(static_cast<ptrdiff_t>(x))
+    #define castFromGlHandle(x) static_cast<unsigned int>(reinterpret_cast<ptrdiff_t>(x))
+
+#else
+
+    #define castToGlHandle(x) (x)
+    #define castFromGlHandle(x) (x)
+
+#endif
```

Comment:
> But in this case, template argument deduction won't work.

It does. Since T is unsigned int, the following happens:
* If GLEXT_GLhandle is unsigned int, then the specialization is used and no conversion happens
* if GLEXT_GLhandle is void* then the generic implementation is used and the conversion is done